src/components/EditorWidgets/List/ListControl.js

```diff
@@ -143,6 +176,13 @@ export default class ListControl extends Component {
     onChange((value || List()).push(parsedValue));
   };
 
+  handleAddType = ({ itemType }) => {
+    const { value, onChange } = this.props;
+    const parsedValue = itemType;
+    this.setState({ itemsCollapsed: this.state.itemsCollapsed.push(false) });
+    onChange((value || List()).push(parsedValue));
```

Comment:
one thing I noticed when trying to use this: the whole of the definition from types is written to the output and you end up with widget and label and the other configuration fields in your formatted file.

this is then read in render, and the widget rendering depends on the whole of the type definition being in the the value object.

ideally this would just store the type name and data in the output, and get the type definition from this.props.types based on the stored type name when rendering.